genienlp/train.py

```diff
@@ -76,40 +76,37 @@ def prepare_data(args, logger):
     train_sets, val_sets, aux_sets, vocab_sets = [], [], [], []
     for task in args.train_tasks:
         logger.info(f'Loading {task.name}')
-        kwargs = {'test': None}
-        kwargs['subsample'] = args.subsample
-        kwargs['validation'] = None
+        kwargs = {'test': None, 'validation': None}
+        kwargs.update({'subsample': args.subsample, 'skip_cache': args.skip_cache,
+                       'cached_path': os.path.join(args.cache, task.name), 'languages': args.train_languages})
         if args.use_curriculum:
             kwargs['curriculum'] = True
-        kwargs['skip_cache'] = args.skip_cache
-        kwargs['cached_path'] = os.path.join(args.cache, task.name)
 
         logger.info(f'Adding {task.name} to training datasets')
         split = task.get_splits(args.data, lower=args.lower, **kwargs)
+        assert not split.eval and not split.test
```

Comment:
This is a very weird way to write this code, basically you have an API that loads 3 datasets at once (train dev test) but you never use it that way, only ever load one dataset at a time.

Reply:
yeah, it is weird, but it only loads the split if it supposed to. Otherwise, it will return None instantly.